verifiers/internal/gha/slsaprovenance/v0.2/byob_test.go

```diff
@@ -0,0 +1,236 @@
+							Environment: map[string]interface{}{
+								"GITHUB_BASE_REF":   "",
+								"GITHUB_REF_TYPE":   "tag",
+								"GITHUB_REF":        "refs/tags/v1.0.0",
+								"GITHUB_EVENT_NAME": "push",
+								"GITHUB_EVENT_PAYLOAD": map[string]any{
+									"base_ref": nil,
+								},
+							},
+						},
+					},
+				},
+			),
+			err: serrors.ErrorInvalidDssePayload,
+		},
+	}
+
+	for i := range testCases {
+		tt := testCases[i]
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			branch, err := tt.prov.GetBranch()
+			if diff := cmp.Diff(tt.err, err, cmpopts.EquateErrors()); diff != "" {
+				t.Fatalf("unexpected error: %v", err)
+			}
+			if got, want := branch, tt.branch; got != want {
+				t.Fatalf("unexpected branch, got: %q, want: %q", got, want)
+			}
+		})
+	}
+}
+
+func Test_byobProvenance_GetTag(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		name string
+		prov iface.Provenance
+		tag  string
+		err  error
+	}{
+		{
+			name: "empty provenance",
+			prov: newBYOBProvenance(
+				&Attestation{
+					StatementHeader: intoto.StatementHeader{},
+					Predicate:       slsa02.ProvenancePredicate{},
+				},
+			),
+			err: serrors.ErrorInvalidDssePayload,
+		},
+		{
+			name: "materials uri @ refs/heads/main",
+			prov: newBYOBProvenance(
+				&Attestation{
+					StatementHeader: intoto.StatementHeader{},
+					Predicate: slsa02.ProvenancePredicate{
+						Materials: []slsacommon.ProvenanceMaterial{
+							{
+								URI: "git+https://github.com/kubernetes/kubernetes@refs/heads/main",
+							},
+						},
+					},
+				},
+			),
+			tag: "",
+		},
+		{
+			name: "environment GITHUB_REF @ refs/heads/main",
+			prov: newBYOBProvenance(
+				&Attestation{
+					StatementHeader: intoto.StatementHeader{},
+					Predicate: slsa02.ProvenancePredicate{
+						Invocation: slsa02.ProvenanceInvocation{
+							Environment: map[string]interface{}{
+								"GITHUB_REF_TYPE": "branch",
+								"GITHUB_REF":      "refs/heads/main",
+							},
+						},
+					},
+				},
+			),
+			err: serrors.ErrorInvalidDssePayload,
+		},
+		{
+			name: "materials uri @ refs/tags/v1.0.0",
+			prov: newBYOBProvenance(
+				&Attestation{
+					StatementHeader: intoto.StatementHeader{},
+					Predicate: slsa02.ProvenancePredicate{
+						Materials: []slsacommon.ProvenanceMaterial{
+							{
+								URI: "git+https://github.com/kubernetes/kubernetes@refs/tags/v1.0.0",
+							},
+						},
+					},
+				},
+			),
+			tag: "refs/tags/v1.0.0",
+		},
+		{
+			name: "environment GITHUB_REF @ ref/tags/v1.0.0",
+			prov: newBYOBProvenance(
+				&Attestation{
+					StatementHeader: intoto.StatementHeader{},
+					Predicate: slsa02.ProvenancePredicate{
+						Invocation: slsa02.ProvenanceInvocation{
+							Environment: map[string]interface{}{
+								"GITHUB_REF_TYPE": "tag",
+								"GITHUB_REF":      "refs/tags/v1.0.0",
+							},
+						},
+					},
+				},
+			),
+			err: serrors.ErrorInvalidDssePayload,
+		},
```

Comment:
add one tests with `URI: "git+https://github.com/kubernetes/kubernetes"` (ie, no branch in resolvedDeps) and 
```json
Environment: map[string]interface{}{
	"GITHUB_REF_TYPE": "tag",
	"GITHUB_REF":      "refs/tags/v1.0.0",
},
```

That's the main common case

Reply:
Done.